README.md

```diff
@@ -38,13 +40,90 @@ class User < ApplicationRecord
 end
 ```
 
-And then it will offer bunch of convenient methods and callbacks that ensure proper state transitions.
+It then will offer a bunch of convenient methods and callbacks that ensure proper state transitions.
+Note the prefix here to prefix the method. This is optional of course.
+
+```ruby
+user = User.new(state: 'active')
+# with the prefix: :state
+user.state_active?  # => true
+# or without the prefix: :state
+user.active? # => true
+
+# The transition check happens when updating the state like this
+user.update!(state: :registered)
+# or when using the shortcut (add state_ because we have prefix: :state above)
+user.state_registered!
+```
+The last command throws an InvalidState error: Invalid transition from active to registered
+This is because the state was not permitted to transition back to "registered" from "active".
+If you do want this, `s.permit_transition(:active, :registered)` should be added.
+
+# Methods
+
 
+## after_inline_transition_to?(to)
+Runs the block inside `after_inline_transition_to` as a before_save action.
+For example the state updates to :registered, but before the model is saved
+```ruby
+state_machine_enum :state, prefix: "state" do |s|
+    s.permit_transition(:registered, :active)
+    s.after_inline_transition_to(:registered) do |model|
+        model.another_attr = Time.now.utc
+    end
+end
 ```
-user = User.new(state: 'registered')
-user.active?
-user.registered! # throws InvalidState error, because state can not transition to "registered".
+`another_attr` is automatically set to the current utc time.
+
+## after_committed_transition_to(to)
+Runs the block inside `after_committed_transition_to` as an after_commit action.
+For example if you want to do something after it has committed to the database when the state is
+updated to :registered
+```ruby
+state_machine_enum :state, prefix: "state" do |s|
+    s.permit_transition(:registered, :active)
+    s.after_committed_transition_to(:registered) do |model|
+        model.send_notification!
+    end
+end
 ```
+
+## after_any_committed_transition
+Runs together with all the `after_committed_transition_to` hooks.
+For example if you want to do something after any state update has commited.
+```ruby
+state_machine_enum :state, prefix: "state" do |s|
+    s.permit_transition(:registered, :active)
+    s.permit_transition(:active, :suspended)
+    s.after_any_committed_transition_to do |model|
+        log_changes!
+    end
+end
+
+```
+
+## Ensure methods
+With a couple of ensure methods we can check beforehand for valid state transitions without actually having to do
+the state transition.
```

Comment:
Suggestion: add "This allows you to bail out of calls where the model is not in a desired state or won't be able to perform a transition, by raising an InvalidState exception"